crates/red_knot_python_semantic/resources/mdtest/call/methods.md

```diff
@@ -0,0 +1,192 @@
+# Methods
+
+## Background: Functions as descriptors
+
+Say we have a simple class `C` with a function definition `f` inside its body:
+
+```py
+class C:
+    def f(self, x: int) -> str:
+        return "a"
+```
+
+Whenever we access the `f` attribute through the class object itself (`C.f`) or through an instance
+(`C().f`), this access happens via the descriptor protocol. Functions are (non-data) descriptors
+because they implement a `__get__` method. This is crucial in making sure that method calls work as
+expected. In general, the signature of the `__get__` method in the descriptor protocol is
+`__get__(self, instance, owner)`. The `self` argument is the descriptor object itself (`f`). The
+passed value for the `instance` argument depends on whether the attribute is accessed from the class
+object (in which case it is `None`), or from an instance (in which case it is the instance of type
+`C`). The `owner` argument is the class itself (`C` of type `Literal[C]`). To summarize:
```

Comment:
it could be useful to link to the Descriptor HOWTO somewhere from this first paragraph (and possibly Brent's blogpost unravelling attribute access too!)

Reply:
Brent? :stuck_out_tongue: 

I linked to [this section](https://docs.python.org/3/howto/descriptor.html#functions-and-methods) in the descriptor guide now. We already refer to the full descriptor guide in `descriptor_protocol.md`, but this section is about `FunctionType.__get__` specifically.

Brett's post doesn't mention bound methods at all, but I'll probably link to that as well when I work on the follow up which distinguishes between `getattr_static`/`Type::static_member` and `find_name_in_mro` :+1: